### PR TITLE
Software backward compatible with CLI 

### PR DESCRIPTION
**Pull request by neel-astro**
## Description
Changes:
- Add logic via high-level function in `cmd/software` package to remove commands based on the Houston version connected
- Add logic via decorator in `houston` package to hide the API based on the Houston version connected
- Add logic to define if varying GraphQL queries based on the Houston version, so that the correct query is picked up based on the Houston version
- Add logic to hide `--use-runtime-version` & `--use-astronomer-certified` flags in `astro dev init` cmd, in case CLI is connected to Houston version < 0.29

Minor Changes:
- Move all the methods in Houston `ClientInterface` to follow a fixed signature `func(req AnyRequestType) (AnyResponseType, error)`, so that the Houston decorator can consume it
- Move Houston GraphQL queries & mutation to individual resource files instead of keeping everything in `queries.go` file, as `queries.go` was becoming quite large
- Add the platform version to the help template in case CLI is connected to Astronomer Software

## 🎟 Issue(s)

Related #633 

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [x] Ran `make lint` before taking out of draft
- [x] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [x] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: cmd/airflow.go [hunk 1]
```diff
@@ -122,16 +123,25 @@ func newAirflowInitCmd() *cobra.Command {
 		RunE: airflowInit,
 	}
 	cmd.Flags().StringVarP(&projectName, "name", "n", "", "Name of Astro project")
-	cmd.Flags().StringVarP(&runtimeVersion, "runtime-version", "v", "", "Specify a version of Astro Runtime that you want to create an Astro project with. If not specified, the latest is assumed. You can change this version in your Dockerfile at any time.")
 	cmd.Flags().StringVarP(&airflowVersion, "airflow-version", "a", "", "Version of Airflow you want to create an Astro project with. If not specified, latest is assumed. You can change this version in your Dockerfile at any time.")
+	var err error
+	var avoidACFlag bool
```
Comment (andriisoldatenko) [inline cmd/airflow.go:128]: can you rename it to something more clear? AC - astronomer certified?

Comment (andriisoldatenko) [reply cmd/airflow.go:128]: may be `avoidACflag`?

Comment (andriisoldatenko) [reply cmd/airflow.go:128]: I think so

**Review by sunkickr — APPROVED** [review]
I checked shared files and they look good to me

**Comment by neel-astro** [comment]
@andriisoldatenko can we have a re-review for this PR, have addressed comments from the first review

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**